src/Core/Checkout/Document/SalesChannel/DocumentRoute.php

```diff
@@ -113,9 +116,17 @@ private function checkAuth(string $documentId, Request $request, SalesChannelCon
             return;
         }
 
-        $this->checkGuestAuth($order, $orderCustomer, $request);
+        if (Feature::isActive('v6.8.0.0')) {
+            // feature flag due to different exceptions
+            $this->guestAuthenticator->validate($order, $request);
+        } else {
+            Feature::silent('v6.8.0.0', fn () => $this->checkGuestAuth($order, $orderCustomer, $request));
+        }
```

Comment:
```suggestion
        // feature flag due to different exceptions
        if (!Feature::isActive('v6.8.0.0')) {
            Feature::silent('v6.8.0.0', fn () => $this->checkGuestAuth($order, $orderCustomer, $request));
            
            return;            
        }
        
        $this->guestAuthenticator->validate($order, $request);
```

Isn't this better for the major update to remove?

Reply:
yes, in the OrderRoute, it's not possible, just in the DocumentRoute, and I copied without thinking 🙈